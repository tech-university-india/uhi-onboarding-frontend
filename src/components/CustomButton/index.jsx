import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

export default function CustomButton ({
  onClick,
  children,
  className = 'bg-purple-500 text-white',
  isSelected = false
}) {
  return (
    <button
      onClick={onClick}

      className={classNames(
        className,
        !isSelected ? 'shadow-inner' : 'shadow-xl',
        'px-8 py-3 rounded-xl text-base font-medium '
      )}
    >
      {children}
    </button>
  )
}

CustomButton.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  isSelected: PropTypes.bool,
  onClick: PropTypes.func
}
