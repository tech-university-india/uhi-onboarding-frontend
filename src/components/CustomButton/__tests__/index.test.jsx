/* eslint-disable no-undef */
import React from 'react'
import CustomButton from '..'
import { fireEvent, render } from '@testing-library/react'
describe('Tests for Custom Button', () => {
  it('should render the button with text', () => {
    const { getByText } = render(<CustomButton>Hello</CustomButton>)
    expect(getByText('Hello')).toBeTruthy()
  })
  it('should show inner shadow when the button is selected', () => {
    const screen = render(<CustomButton isSelected>Hello</CustomButton>)
    expect(screen.getByText('Hello').className).toContain('shadow-selected')
  })
  it('should show outer shadow when the button is not selected', () => {
    const screen = render(<CustomButton>Hello</CustomButton>)
    expect(screen.getByText('Hello').className).toContain('shadow-xl')
  })
  it('should call the onClick function when the button is clicked', () => {
    const onClick = jest.fn()
    const screen = render(<CustomButton onClick={onClick}>Hello</CustomButton>)
    fireEvent.click(screen.getByText('Hello'))
    expect(onClick).toHaveBeenCalled()
  })
})
