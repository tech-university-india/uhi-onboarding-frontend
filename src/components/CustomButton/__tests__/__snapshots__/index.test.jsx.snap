// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <button
    class="shadow-xl px-6 py-3 rounded-xl text-base font-medium "
    style="background-color: rgb(229, 252, 244);"
  >
    Hello
  </button>
</div>
`;
